server/course/router.go

```
@@ -62,7 +62,7 @@ func getCourseByID(c *gin.Context) {
 		return
 	}
 
-	hasAccess, err := permissionValidation.CheckCoursePermission(c, id, []string{"Lecturer", "Editor", "Student"}, "courses:view-all")
+	hasAccess, err := permissionValidation.CheckCoursePermission(c, id, []string{keycloak.PromptAdmin, keycloak.CourseLecturer, keycloak.CourseEditor, keycloak.CourseStudent})
 	if err != nil || !hasAccess {
 		return
```

Maybe return a proper Error code here